cloud/services/compute/loadbalancers/reconcile.go

```diff
@@ -130,60 +298,94 @@ func (s *Service) createOrGetInstanceGroups(ctx context.Context) ([]*compute.Ins
+	healthcheckSpec := s.scope.HealthCheckSpec(lbname)
 	log.V(2).Info("Looking for healthcheck", "name", healthcheckSpec.Name)
-	healthcheck, err := s.healthchecks.Get(ctx, meta.GlobalKey(healthcheckSpec.Name))
+	key := meta.GlobalKey(healthcheckSpec.Name)
+	healthcheck, err := s.healthchecks.Get(ctx, key)
 	if err != nil {
 		if !gcperrors.IsNotFound(err) {
 			log.Error(err, "Error looking for healthcheck", "name", healthcheckSpec.Name)
 			return nil, err
 		}
 
 		log.V(2).Info("Creating a healthcheck", "name", healthcheckSpec.Name)
-		if err := s.healthchecks.Insert(ctx, meta.GlobalKey(healthcheckSpec.Name), healthcheckSpec); err != nil {
+		if err := s.healthchecks.Insert(ctx, key, healthcheckSpec); err != nil {
 			log.Error(err, "Error creating a healthcheck", "name", healthcheckSpec.Name)
 			return nil, err
 		}
 
-		healthcheck, err = s.healthchecks.Get(ctx, meta.GlobalKey(healthcheckSpec.Name))
+		healthcheck, err = s.healthchecks.Get(ctx, key)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return healthcheck, nil
+}
+
+func (s *Service) createOrGetRegionalHealthCheck(ctx context.Context, lbname string) (*compute.HealthCheck, error) {
+	log := log.FromContext(ctx)
+	healthcheckSpec := s.scope.HealthCheckSpec(lbname)
+	healthcheckSpec.Region = s.scope.Region()
+	log.V(2).Info("Looking for regional healthcheck", "name", healthcheckSpec.Name)
+	key := meta.RegionalKey(healthcheckSpec.Name, s.scope.Region())
+	healthcheck, err := s.regionalhealthchecks.Get(ctx, key)
+	if err != nil {
+		if !gcperrors.IsNotFound(err) {
+			log.Error(err, "Error looking for regional healthcheck", "name", healthcheckSpec.Name)
+			return nil, err
+		}
+
+		log.V(2).Info("Creating a regional healthcheck", "name", healthcheckSpec.Name)
+		if err := s.regionalhealthchecks.Insert(ctx, key, healthcheckSpec); err != nil {
+			log.Error(err, "Error creating a regional healthcheck", "name", healthcheckSpec.Name)
+			return nil, err
+		}
+
+		healthcheck, err = s.regionalhealthchecks.Get(ctx, key)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	s.scope.Network().APIServerHealthCheck = ptr.To[string](healthcheck.SelfLink)
 	return healthcheck, nil
 }
 
-func (s *Service) createOrGetBackendService(ctx context.Context, instancegroups []*compute.InstanceGroup, healthcheck *compute.HealthCheck) (*compute.BackendService, error) {
+func (s *Service) createOrGetBackendService(ctx context.Context, lbname string, mode loadBalancingMode, instancegroups []*compute.InstanceGroup, healthcheck *compute.HealthCheck) (*compute.BackendService, error) {
 	log := log.FromContext(ctx)
 	backends := make([]*compute.Backend, 0, len(instancegroups))
 	for _, group := range instancegroups {
-		backends = append(backends, &compute.Backend{
-			BalancingMode: "UTILIZATION",
+		be := &compute.Backend{
+			BalancingMode: string(mode),
 			Group:         group.SelfLink,
-		})
+		}
+		if mode == loadBalancingModeConnection {
+			be.MaxConnections = int64(2 ^ 32)
```

Comment:
Can we add a comment specifying where this value is coming from?

Reply:
Good point, this came from an internal implementation but I'll update it with a clarifying comment. TBD.